libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -21,15 +21,19 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  *
  */
-
 #include <Arduino.h>
+
+#include "ESP8266HTTPClient.h"
+
+#ifdef HTTPCLIENT_1_1_COMPATIBLE
```

Comment:
Same concern as before.  If you use `#ifdef XXXX` then I think you're going the wrong way.  Both APIs should be available, with no need for `#ifdef`s.  Right now you either compile an object with one API or with the other API, but never both.  So even basic compile checks aren't being done on one code path (unless you explicitly make an older test case and `#undef` as appropriate...

@igrr and @devyte , am I making sense or is this the way you'd do the migration?  Don't want to harp on something that may not be a problem.

Reply:
Sure changing the name does not change the concept. I tried to explain why I did this though. With the #define HTTPCLIENT_1_1_COMPATIBLE both APIs, deprecated and new, are available. In the next release everything between #ifdef .. #endif can be removed, including the #define itself. Only the new functions remain.
Don't worry about the basic compiler checks, there is only one #else to make the two variants of the end() function more readable. If you prefer to have no #else I can fix that. Without any #else only #ifdefs and the #define *everything* is compiled...